recipe/meta.yaml

```diff
@@ -332,6 +336,8 @@ outputs:
     build:
       skip: true  # [python_impl != 'cpython']
     requirements:
+      build:
+        - {{ stdlib("c") }}
```

Comment:
@h-vetinari, @beckermr, this one definitely isn't right since we have a `script`-less (i.e., dependency-only "meta-package")  output for which we shouldn't add any additional build requirements and -- importantly -- `run_exports` coming from those then.

Reply:
This is a bit trickier. There's some logic in the piggyback that basically says: if we use it in the global build environment, we might need it for all the outputs, so by default we add it defensively in such cases.

Importantly, it doesn't hurt in this case AFAICT because it does not add any transitive run-deps (from _using_ this package), and the run-exports on this outputs itself are already implied anyway by the explicit dependence on other subpackages that require it.

In any case, if you think this should be handled better/differently/less defensively and have a smart way to detect these cases, I'm happy to discuss how to implement it in the piggyback migrator!

Reply:
> In any case, if you think this should be handled better/differently/less defensively

It's just that my brain, as usual, jumps to enumerating corner cases when I see such changes. Like, would this affect cases with `noarch` outputs, etc.

> and have a smart way to detect these cases, I'm happy to discuss how to implement it in the piggyback migrator!

I don't think I'll be able to find time to review/advice on the migrator code, sorry.
I just wanted to raise awareness since encountering these only partially handled cases early in the migration gives an indication that we have a higher probability to see more of those.

Reply:
I'm of the opinion that there's only so much we can do with the piggyback. But it would be very easy to adapt the migrator to not add a build section to a given output, if that output doesn't already have a build section.

Reply:
> this one definitely isn't right since we have a `script`-less (i.e., dependency-only "meta-package") output

OK, the `script`-less part gives something to hook onto here. I'll look at skipping outputs that have no script and no build section.